transport/http/handler.go

```diff
@@ -144,8 +143,12 @@ func handleOnewayRequest(
 	}
 	treq.Body = &buff
 
+	// create a new context for oneway requests since the HTTP handler cancels
+	// http.Request's context when ServeHTTP returns
+	ctx := opentracing.ContextWithSpan(context.Background(), span)
```

Comment:
the semantics of a oneway http request are a little odd. What does guaranteed delivery mean in this case? What happens if this context’s deadline is exceeded or ends with another error?

This perhaps highlights why HTTP is a poor fit for oneway. Async queueing should not be a black hole.